### PR TITLE
Make client requests by URI

### PR DESCRIPTION
**Pull request by djmetzle**
Make the uri object for endpoints first class.

Add a class `ApiUri` that handles putting together the full endpoint
urls. This includes GET params.

Modify the client class to accept these full uri objects when making
requests.

### CODE REVIEW

File: lib/client/client.rb [hunk 1]
```diff
@@ -4,40 +4,37 @@
 require "json"
 
 class ApiClient
-   ENDPOINT = "https://api.replicated.com/vendor/v1"
+   VENDOR_ENDPOINT = "https://api.replicated.com"
```
Comment (g1y) [inline lib/client/client.rb]: vendor/v1?

Comment (g1y) [reply lib/client/client.rb]: nvm, we are only using the host and port. May be better to just change this to `VENDOR_HOST` and add a `VENDOR_PORT` const

**Comment by g1y** [comment]
CR :+1:

**Comment by djmetzle** [comment]
Mergin' it.

**Pull request merged**